### PR TITLE
docs: improve UDF signature docs

### PR DESCRIPTION
**Pull request by NickCrews**
Also added callout to optional args signature for builtin UDFs. Didn't for the other UDF implementations.

Fixes https://github.com/ibis-project/ibis/issues/8170#issuecomment-1921802611

### CODE REVIEW

**Comment by lostmygithubaccount** [comment]
the docs build error means you didn't check in (after rendering/previewing) the frozen execution outputs for the doc, so it's trying to run them

**Comment by cpcloud** [comment]
It looks like we are running snowflake queries in this part of the docs so for now someone with credentials will need to run it. In the meantime we should ideally move the snowflake stuff to a different document or remove it.

**Comment by NickCrews** [comment]
(not looking at the code as I say this) can we just switch the examples to duckdb or something the doesn't need creds?

**Comment by NickCrews** [comment]
Does the rest of the changes look good though?

File: docs/how-to/extending/builtin.qmd [hunk 1]
```diff
@@ -164,7 +161,9 @@ con.execute(cardinality([1, 2, 3]))
 con.execute(cardinality(["a", "b"]))
 ```
 
-When you bypass input types the errors you get back are backend dependent:
+When you skip giving input types, Ibis isn't able to catch your typing errors
```
Comment (cpcloud) [inline docs/how-to/extending/builtin.qmd:164]:
```suggestion
When you do not specify input types, Ibis isn't able to catch typing errors
```

**Comment by NickCrews** [comment]
I removed/ported the snowflake stuff. we cab revert that if you want, but I think this is an improvement.

File: docs/how-to/extending/builtin.qmd [hunk 2]
```diff
@@ -164,14 +116,16 @@ con.execute(cardinality([1, 2, 3]))
...
+early, and they are only caught during execution.
+The errors you get back are backend dependent:
 
 ```{python}
 #| error: true
 con.execute(cardinality("foo"))
 ```
 
-Here, Snowflake is informing us that the `ARRAY_SIZE` function does not accept
+Here, Duckdb is informing us that the `ARRAY_LENGTH` function does not accept
```
Comment (cpcloud) [inline docs/how-to/extending/builtin.qmd:128]:
```suggestion
Here, DuckDB is informing us that the `ARRAY_LENGTH` function does not accept
```

**Comment by cpcloud** [comment]
@NickCrews Nah, let's keep it, this is strictly an improvement! Thank you!

**Review by cpcloud — APPROVED** [review]

**Comment by cpcloud** [comment]
LGTM: https://pr-8194-b47c22b9f167a7f19b7670f18528cf8848007ad7--ibis-quarto.netlify.app/how-to/extending/builtin

Thanks again!

**Pull request merged**